include/insns/RV32I.h

```diff
@@ -249,7 +239,8 @@ class RV32I : public RevExt {
     { RevInstDefaults().SetMnemonic("sub %rd, %rs1, %rs2"  ).SetFunct3(   0b000).SetImplFunc(sub   ).SetFunct2or7(0b0100000)                                                  .SetFormat(RVTypeR).SetOpcode(0b0110011) },
     { RevInstDefaults().SetMnemonic("sll %rd, %rs1, %rs2"  ).SetFunct3(   0b001).SetImplFunc(sll   ).SetFunct2or7(0b0000000)                                                  .SetFormat(RVTypeR).SetOpcode(0b0110011) },
     { RevInstDefaults().SetMnemonic("slt %rd, %rs1, %rs2"  ).SetFunct3(   0b010).SetImplFunc(slt   ).SetFunct2or7(0b0000000)                                                  .SetFormat(RVTypeR).SetOpcode(0b0110011) },
-    { RevInstDefaults().SetMnemonic("sltu %rd, %rs1, %rs2" ).SetFunct3(   0b011).SetImplFunc(sltu  ).SetFunct2or7(0b0000000)                                                  .SetFormat(RVTypeR).SetOpcode(0b0110011) },
+    { RevInstDefaults().SetMnemonic("sltu %rd, %rs1, %rs2" ).SetFunct3(   0b011).SetImplFunc(sltu  ).SetFunct2or7(0b0000000)                                                  .SetFormat(RVTypeR).SetOpcode(0b0110011).SetPredicate( []( uint32_t Inst ){ return DECODE_RS1( Inst ) != 0; } ) },
+    { RevInstDefaults().SetMnemonic("snez %rd, %rs"        ).SetFunct3(   0b011).SetImplFunc(sltu  ).SetFunct2or7(0b0000000)                                                  .SetFormat(RVTypeR).SetOpcode(0b0110011).SetPredicate( []( uint32_t Inst ){ return DECODE_RS1( Inst ) == 0; } ) },
```

Comment:
Do we need an implementation of snez or does it fall out of the sltu implementation?

Reply:
> Do we need an implementation of snez or does it fall out of the sltu implementation?

`snez` is a pseudoinstruction which is a `sltu` instruction.

All pseudoinstructions listed in the RISC-V spec were implemented by me, by using a shorter name for the same instruction where applicable.

We do not have "an implementation of snez". We have a textual disassembly of `sltu` as `snez` when it applies as the pseudoinstruction mnemonic. The functionality is the same, provided by the `sltu` execute function.

Reply:
> We do not have "an implementation of snez". We have a textual disassembly of `sltu` as `snez` when it applies as the pseudoinstruction mnemonic. The functionality is the same, provided by the `sltu` execute function.

I think the source of confusion may be that in the past, there was one _and only one_ instruction table entry for each instruction, and that instruction necessarily had its own, and only, implementation.

The new table is not a one-to-one mapping anymore, hence the use of `std::unordered_multimap` instead of `std::map`. Instead, multiple table entries can exist for the same encoding, and only the one whose predicate is true is chosen, such as the pseudoinstruction mnemonic for an otherwise general instruction.

`addi` has 3 table entries, one for the general `addi` instruction, one for the `mv` class of pseudoinstructions, and one for the very specific pseudoinstruction `nop`. They all share the same `addi` execute function -- we do not need to implement `mv` or `nop` as separate execute functions, since they are all part of the same `addi` instruction. But we can distinguish them by their displayed mnemonic when we disassemble/trace them.

Reply:
makes sense - thanks Lee